### ISSUE TITLE
Use .gitmodules to automatically define submodules

### ISSUE THREAD

**Issue by Fmstrat** [mhutchie/vscode-git-graph#173]
**Describe the improvement that you'd like**
Similar to the request in https://github.com/mhutchie/vscode-git-graph/issues/155

Git Graph is pretty amazing, but for those of us in companies with large monorepos it's difficult to use because of the volume of submodules. Unfortunately manually adding them is far too tedious to do for users when there may be dozens.

The reason detection wasn't implemented as defined in 155 was:

> The reason why they cannot be automatically scanned is due to the exponential search cost this adds when the extension is searching for repos (that occurs every time Visual Studio Code is opened), as the extension would have to traverse the entire directory structure (on large projects and workspaces this is extremely slow). 

My suggestion is to leverage the `.gitmodules` file to identify submodules, and add them in, similar to how the Git plugin for VSCode works. The file format is:
```
[submodule "stacks/mystack/repos/app"]
        path = stacks/mystack/repos/app
        url = git@ssh.dev.site.com:v3/myuser/app
```
The path/url variables could then be used to define submodules at load time.

Thanks!

**Comment by mhutchie** [comment]
Thanks for raising this improvement request!

I wasn't aware of `.gitmodules`, thanks for letting me know! From the information you've provided, and from the git documentation I've just been reading, it appears to contain all of the necessary information for Git Graph to automatically add the defined submodules (without introducing significant cost for their discovery).

I'll add this in an upcoming release.

**Comment by mhutchie** [comment]
This will be available in v1.15.0.

If you'd like to use it before the next release, you can download [v1.15.0-beta.1](https://github.com/mhutchie/vscode-git-graph/releases/tag/v1.15.0-beta.1), and install it following the instructions provided [here](https://github.com/mhutchie/vscode-git-graph/wiki/Downloading-&-Installing-Releases#manual-installation).
